Etabs_Adapter/CRUD/Create/SectionProperty.cs

```diff
@@ -238,5 +241,167 @@ private void SetProfile(IProfile profile, string sectionName, IMaterialFragment
         }
 
         /***************************************************/
+
+
+        private bool GetFromDatabase(ISectionProperty bhSection)
+        {
+            if (EtabsSettings.DatabaseSettings.SectionDatabase == SectionDatabase.None)
+                return false;
+
+            string bhName = bhSection.Name;
+
+            bhName = bhName.ToUpper();
+            bhName = bhName.Replace(" ", "");
+
+            string[] sub = bhName.Split('X');
+
+            for (int i = 0; i < sub.Length; i++)
+            {
+                if (sub[i].EndsWith(".0"))
+                    sub[i] = string.Join("", sub[i].Take(sub[i].Length - 2).ToArray());
+            }
+
+            bhName = string.Join("X", sub);
+
+            string startOfName = string.Concat(bhName.TakeWhile(c => c < '0' || c > '9'));
+            string replace;
+            if (GetDictionary(EtabsSettings.DatabaseSettings.SectionDatabase).TryGetValue(startOfName, out replace))
+                bhName = bhName.Replace(startOfName, replace);
+
+            int index = m_DBSectionsNames.Select(x => x.ToUpper().Replace(" ", "")).ToList().IndexOf(bhName);
+            if (index == -1)
+                return false;
+
+            if (1 == m_model.PropFrame.ImportProp(
+                                                bhSection.Name,
+                                                bhSection.Material.Name,
+                                                EnumToString(EtabsSettings.DatabaseSettings.SectionDatabase),
+                                                m_DBSectionsNames[index]))
+            {
+                return false;
+            }
+
+            Engine.Reflection.Compute.RecordNote(bhSection.Name + " properties has been assigned from the database section " + bhName + ".");
+            return true;
+        }
+
+        /***************************************************/
+
+        private Dictionary<string, string> GetDictionary(SectionDatabase sectionDatabase)
```

Comment:
Would rename this method to something more explicit to what it actually is doing. Even something longer like `BHoMToEtabsNamingConventions` (or shorter if you can think of something better) but at least something explaining what the method is doing.